### PR TITLE
Add profile event in hedged requests and change timeout from sec to ms.

### PR DESCRIPTION
**Pull request by Avogar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add profile event HedgedRequestsChangeReplica, change read data timeout from sec to ms.

Detailed description / Documentation draft:
HedgedRequestsChangeReplica event counts the total number of times when a change replica timeout has been alarmed. Change `read_data_timeout` timeout from sec to ms, add `_ms` ending to the timeouts. Add better processing the case when `max_parallel_replicas` > 1 but there is no way to sample data in this query.

### CODE REVIEW

File: tests/integration/test_hedged_requests_parallel/configs/users.xml [hunk 1]
```diff
@@ -4,8 +4,8 @@
         <default>
             <load_balancing>in_order</load_balancing>
             <max_parallel_replicas>2</max_parallel_replicas>
-            <hedged_connection_timeout>100</hedged_connection_timeout>
-            <receive_data_timeout>2</receive_data_timeout>
+            <hedged_connection_timeout_ms>100</hedged_connection_timeout_ms>
```
Comment (akuzm) [inline tests/integration/test_hedged_requests_parallel/configs/users.xml:7]: This one should be increased, right?

**Review by akuzm — APPROVED** [review]

File: tests/integration/test_hedged_requests_parallel/configs/users.xml (hunk 1 above)
Comment (alexey-milovidov) [reply tests/integration/test_hedged_requests_parallel/configs/users.xml:7]: Connection timeout was already in ms.

**Comment by Avogar** [comment]
> Many integration tests from test_distributed_respect_user_timeouts are flaky under tsan, can this be related?
> 
> https://clickhouse-test-reports.s3.yandex.net/21886/055242fe95d8f8750355fe68d6453e0abc305957/integration_tests_(thread).html#fail1

I see the same problem on master, I think it's not related to this PR.

**Comment by akuzm** [comment]
OK, let's merge then.

**Pull request merged**